docs/Migration Guide.md

```diff
@@ -0,0 +1,152 @@
+# Migration Guide #
+
+This document describes how to port your code from using one ESR version
+of SpiderMonkey to the next ESR version.
+
+## ESR 60 to ESR 68 ##
+
+### Remove requests ###
+
+Previously, a "request" was a global lock on the JS interpreter that was
+required in order to use many parts of the JSAPI.
+This is no longer necessary, and can simply be deleted.
+
+**Recommendation**: Remove all `JSAutoRequest`, `JS_BeginRequest()`, and
+`JS_EndRequest()` from your code.
+
+### Rename compartments to realms ###
+
+In ESR 68, a "realm" is the term for the environment belonging to a
+particular global object, so when you enter the "realm" of a global
+object you are executing code with that object as the global object.
+As before, with compartments, objects from different realms cannot
+access each other unless they are "wrapped".
```

Comment:
Same-compartment different-realm objects do *not* need to be wrapped. The first sentence is still correct.

Objects in different compartments still need be wrapped. You will want to enter the appropriate realm, and there's less checking if you're doing it the right way when accessing objects from different realms (unless you run with the equivalent of the JS shell's --more-compartments, which I think does compartment-per-realm.)